layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -559,30 +559,138 @@ in a split window to the right."
   (split-window-horizontally)
   (other-window 1))
 
-(defun spacemacs/layout-grid ()
-  " Set the layout to a 2x2 grid. "
-  (interactive)
-  (delete-other-windows)
-  (split-window (split-window-right) nil 'below)
-  (split-window-below))
-
-(defun spacemacs/layout-triple-columns ()
-  " Set the layout to triple columns. "
-  (interactive)
-  (delete-other-windows)
-  (dotimes (i 2) (split-window-right))
+
+;; Window Split
+
+(defun spacemacs--window-split-non-ignored-windows ()
+  "Determines the list of windows to be deleted."
+  (seq-filter
+    (lambda (window)
+      (let* ((name (buffer-name (window-buffer window)))
+              (prefixes-matching
+                (seq-filter
+                  (lambda (prefix) (string-prefix-p prefix name))
+                  spacemacs-window-split-ignore-prefixes)))
+        (not prefixes-matching)))
+    (window-list (selected-frame))))
+
+(defun spacemacs/window-split-default-delete ()
+  "Deletes other windows, except a list of excluded ones."
+  (if spacemacs-window-split-ignore-prefixes
+      (let* ((to-delete (spacemacs--window-split-non-ignored-windows)))
+        (when to-delete
+          (select-window (car to-delete))
```

Comment:
If `(car to-delete)` is a side window, could face errors as noted in comment below. Maybe select the first window from `to-delete` that isn't a side window?